packages/expo-camera/src/CameraModule/CameraModule.ts

```diff
@@ -123,66 +125,83 @@ class CameraModule {
     this.videoElement = videoElement;
     if (this.videoElement) {
       this.videoElement.addEventListener('loadedmetadata', () => {
-        this._syncTrackCapabilities();
+        this.syncTrackCapabilities();
       });
     }
   }
 
-  async onCapabilitiesReady(track: MediaStreamTrack): Promise<void> {
-    const capabilities = track.getCapabilities() as any;
+  public isTorchAvailable(): boolean {
+    return isCapabilityAvailable(this.videoElement, 'torch');
+  }
+
+  public isZoomAvailable(): boolean {
+    return isCapabilityAvailable(this.videoElement, 'zoom');
+  }
+
+  private async onCapabilitiesReady(track: MediaStreamTrack): Promise<void> {
+    const capabilities: any = track.getCapabilities();
 
     // Create an empty object because if you set a constraint that isn't available an error will be thrown.
     const constraints: {
       zoom?: number;
       torch?: boolean;
       whiteBalance?: string;
       focusMode?: string;
+      height?: number;
+      width?: number;
+      aspectRatio?: number;
     } = {};
 
     if (capabilities.zoom) {
-      // TODO: Bacon: We should have some async method for getting the (min, max, step) externally
       const { min, max } = capabilities.zoom;
-      constraints.zoom = Math.min(max, Math.max(min, this._zoom));
+      const converted = convertRange(this._zoom, [min, max]);
+      constraints.zoom = Math.min(max, Math.max(min, converted));
     }
+
     if (capabilities.focusMode) {
       constraints.focusMode = CapabilityUtils.convertAutoFocusJSONToNative(this.autoFocus);
     }
+
     if (capabilities.torch) {
       constraints.torch = CapabilityUtils.convertFlashModeJSONToNative(this.flashMode);
+      console.log('constraints.torch', constraints.torch)
     }
     if (capabilities.whiteBalance) {
       constraints.whiteBalance = this.whiteBalance;
     }
 
+    // Create max-res camera
+    // if (capabilities.aspectRatio && capabilities.aspectRatio.max) {
+    //   constraints.aspectRatio = capabilities.aspectRatio.max;
+    // }
+
     await track.applyConstraints({ advanced: [constraints] as any });
   }
 
-  async _syncTrackCapabilities(): Promise<void> {
-    if (this.stream) {
-      await Promise.all(this.stream.getTracks().map(track => this.onCapabilitiesReady(track)));
+  private async applyVideoConstraints(constraints: { [key: string]: any }): Promise<boolean> {
```

Comment:
Leaving this here to simplify future PRs related to video and audio saving. Otherwise I'll be merging the complexity of those PRs (web workers) with the cross-browser logic utilities.